uni/lib/view/exams/widgets/exam_filter_form.dart

```diff
@@ -60,10 +70,10 @@ class ExamFilterFormState extends State<ExamFilterForm> {
             maxLines: 2,
           ),
           key: Key('ExamCheck$key'),
-          value: filteredExams[key],
+          value: widget.filteredExams[key],
           onChanged: (value) {
             setState(() {
-              filteredExams[key] = value!;
+              widget.filteredExams[key] = value!;
```

Comment:
Notice that you are not changing the widget's state, but rather the widgets properties. It might be working since setState does trigger a rebuild of the entire widget, but I believe there's a better way to do this.
Ideally, the widget should not change outside values by aliasing, but through a callback. How about:
- The widget receives the initial filtered list via an immutable list view
- The widget mutates it via a callback